pkg/storage/cacher.go

```diff
@@ -111,10 +111,56 @@ type Cacher struct {
 	ListFromCache bool
 }
 
+type DecorateStorage func(
+	Interface, int, Versioner, runtime.Object, string, bool,
+	func() runtime.Object) Interface
```

Comment:
Can you name the parameters? + add godoc.

I might call this a storage factory...